### PR TITLE
feat: new node modal component

### PR DESCRIPTION
**Pull request by Amyh102**
New node/graph component
Screen Shot 2022-03-29 at 12 04 07 AM

### CODE REVIEW

File: src/components/modals/NewNodeModal/NewNodeModal.tsx [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+          createNodeInput: {
+            data: JSON.stringify({
+              ...defaultNodeData,
+              title: title,
+              description: description,
+            }),
+          },
+        },
+      })
+      push(`/`)
```
Comment (richardxhong) [inline src/components/modals/NewNodeModal/NewNodeModal.tsx]: we certainly dont want to redirect them back to the main page

**Review by richardxhong — CHANGES_REQUESTED** [review]
99% of the changes seem to be a duplicate of the NewProjectModal

Perhaps we shouldn't duplicate the icons, and styles?

File: src/components/modals/NewNodeModal/NewNodeModal.tsx [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+    const title = data.graphName
+    const description = data.description
+    try {
+      await createNode({
+        variables: {
+          createNodeInput: {
+            data: JSON.stringify({
+              ...defaultNodeData,
+              title: title,
+              description: description,
```
Comment (richardxhong) [inline src/components/modals/NewNodeModal/NewNodeModal.tsx]: the createNodeInput should be of form: 
```ts
{
  projectId: Int
  description: string
  nodeType: string
}
```

u passed in 

```
{
  data: string
}
```

Comment (Amyh102) [reply src/components/modals/NewNodeModal/NewNodeModal.tsx]: I think title and description are in the data part, just updated the useCreateNodeMutation

File: src/components/modals/NewNodeModal/NewNodeModal.tsx [hunk 3]
```diff
@@ -0,0 +1,176 @@
+/* eslint-disable jsx-a11y/no-static-element-interactions */
+/* eslint-disable jsx-a11y/click-events-have-key-events */
+import { ChangeEvent, useCallback, useContext, useState } from 'react'
+
+import { ModalWrapper } from 'src/components/wrappers/ModalWrapper'
+import { TextField, Button, BarGraphNode, PieGraphNode } from 'src/components'
+import { close, bar, pie, scatter, line } from '../NewProjectModal/icons'
+
+import styles from '../NewProjectModal/NewProjectModal.module.scss'
+import style from './NewNodeModal.module.scss'
```
Comment (richardxhong) [inline src/components/modals/NewNodeModal/NewNodeModal.tsx]:
```suggestion
import newModalStyles from './NewNodeModal.module.scss'
```

File: src/components/modals/NewNodeModal/NewNodeModal.tsx [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+  graphType: 'BAR',
+}
+
+export const NewNodeModal = () => {
+  const [data, setData] = useState({ ...initialData })
+  const {
+    useNewNodeModalState: [, setNewNodeModalState],
+  } = useContext(ModalContext)
+  const [createNode] = useCreateNodeMutation()
+  const { push } = useRouter()
```
Comment (richardxhong) [inline src/components/modals/NewNodeModal/NewNodeModal.tsx]:
```suggestion
```

not used.

File: src/components/modals/NewNodeModal/NewNodeModal.tsx [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+import { useRouter } from 'next/router'
+
+const initialData = {
+  graphName: '',
+  description: '',
+  graphType: 'BAR',
+}
+
+export const NewNodeModal = () => {
+  const [data, setData] = useState({ ...initialData })
```
Comment (richardxhong) [inline src/components/modals/NewNodeModal/NewNodeModal.tsx]:
```suggestion
  const [data, setData] = useState(initialData)
```

**Review by richardxhong — CHANGES_REQUESTED** [review]

**Pull request closed without merging**